### PR TITLE
Condense OMP, Fix Correctness Bug: OCC's Open-Heart Surgery IV

### LINKED ISSUE #1808: OMP2.5 Amplitudes Not Converging
Upon examining the reference output from tests, none of the four OMP2.5 tests show convergence of the T2 amplitudes. [The T2 residuals stay at the same order of magnitude for every single iteration.](https://github.com/psi4/psi4/blob/master/tests/omp2p5-2/output.ref#L352-L360) Neither the OMP3 tests nor the DF-OMP2.5 tests show this behavior, which suggests the T2 amplitude non-convergence is a bug in the `occ` modification of OMP3 to produce OMP2.5.

This issue blocks improving orbital convergence in OCC by coupling it to amplitude convergence, issue #1777. Due to this issue, the proposed scheme instead causes non-convergence of both quantities. I will fix this issue as the next PR in the "OCC's Open-Heart Surgery" series. Core developers, please mark this issue as a 1.4 target.

### PR DESCRIPTION
**Pull request by JonathonMisiewicz**
## Description
While working on a fix to #1777, I encountered bug #1808, which blocks my fix. This PR fixes #1808 and condenses some of the relevant code.

The bug is discussed in detail at https://github.com/psi4/psi4/issues/1808. Briefly, OMP2.5 theory has two sets of amplitudes, T2(1) and T2(2). Compared to the very similar OMP3 theory, it halves certain terms. OCC implemented this by halving the T2(2) amplitudes instead of halving intermediates that use them. While the intermediates will be correct, this has the side-effect of changing the T2(2) amplitudes from their current values, so they do not converge. This PR changes the implementation so the necessary terms are halved, but not the T2(2) amplitudes. Now they converge properly and agree better with my Psi4Numpy implementation of OMP2.5.

Disclaimer that @bozkaya has yet to comment. While I'm confident I've implemented the theory correctly, he is more familiar with OMP2.5 than I am.

## Todos
Notable points (developer or user-interest) that this PR has or will accomplish.
- [x] Fixes a bug causing small (~fifth decimal place) errors in OMP2.5 gradients with conventional integrals.
- [x] Condenses OMP2 and OMP2.5/OMP3's functions for the first order T2 amplitudes. 

## Checklist
- [x] Quick tests pass. `^omp2p5` passes. `^omp3` passes with the exceptions noted in #1793. 

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: psi4/src/psi4/occ/omp3_g_int.cc [hunk 1]
```diff
@@ -274,59 +257,23 @@ void OCCWave::omp3_g_int() {
...
-        global_dpd_->file2_mat_init(&G);
-        global_dpd_->file2_mat_rd(&G);
-#pragma omp parallel for
-        for (int h = 0; h < nirrep_; ++h) {
-            for (int i = 0; i < aoccpiA[h]; ++i) {
-                for (int j = 0; j < aoccpiA[h]; ++j) {
-                    GooA->set(h, i, j, G.matrix[h][i][j]);
-                }
-            }
-        }
```
Comment (dgasmith) [inline psi4/src/psi4/occ/omp3_g_int.cc]: Nice!

**Comment by JonathonMisiewicz** [comment]
Re-pinging @bozkaya. Even an estimate on when you can look at this would be helpful.

**Comment by bozkaya** [comment]
You may verify your changes with respect to DFOCC CD-OMP2.5 code. In the case of energy with CD tolerance of 10^-14, CD becomes exact. Hence, you may verify your changes with respect to DFOCC module. As long as you can reproduce my results in OMP2.5 paper (with CDS) or you can get the same results with CD-OMP2.5, you are welcome to make any change in OCC module. Frankly speaking, I have focused on DFOCC since 2014. @JonathonMisiewicz @loriab

**Comment by JonathonMisiewicz** [comment]
The CD test passes, so we're clear to merge once pending reviews are in.

**Comment by bozkaya** [comment]
Then, your changes are okay. Thank you for your effort! @JonathonMisiewicz

**Pull request merged**